[macOS] Temporarily disable JPEG XL on older OSes until we can get https://bugs.webkit.org/show_bug.cgi?id=259587 ironed out

litherum opened the pull request:
#### 4ef6f54430f8a53c3de39ebc57edc245f304e620
<pre>
[macOS] Temporarily disable JPEG XL on older OSes until we can get <a href="https://bugs.webkit.org/show_bug.cgi?id=259587">https://bugs.webkit.org/show_bug.cgi?id=259587</a> ironed out
<a href="https://bugs.webkit.org/show_bug.cgi?id=259603">https://bugs.webkit.org/show_bug.cgi?id=259603</a>
rdar://113041105

Reviewed by Tim Horton.

This path is producing images that are garbled. Temporarily disable the feature until we can fix that.

* Source/WebCore/Configurations/WebCore.xcconfig:

Canonical link: <a href="https://commits.webkit.org/266408@main">https://commits.webkit.org/266408@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/42e044c2cf13c6051ff0ef53415540b03b1ad341

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/13721 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/14035 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/14368 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/15457 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/13029 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/13802 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/16543 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/14116 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/15708 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/13887 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/14508 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/11615 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/16156 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/11797 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/12373 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/19407 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/12872 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/12538 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/15750 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/13066 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/10943 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/12334 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/3345 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/16667 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/12907 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

brentfulgham commented on Source/WebCore/Configurations/WebCore.xcconfig:
```diff
@@ -26,9 +26,7 @@
-WK_USE_APPLEJPEGXL = $(WK_USE_APPLEJPEGXL_$(WK_PLATFORM_NAME));
-WK_USE_APPLEJPEGXL_macosx = $(WK_USE_APPLEJPEGXL$(WK_MACOS_1400));
-WK_USE_APPLEJPEGXL_MACOS_BEFORE_1400 = YES;
+WK_USE_APPLEJPEGXL = NO;
```
Won't this turn JPEGXL off for all builds? Is this mean to go to a branch only?

brentfulgham replied:
@emw-apple Should this be something like:

```
WK_USE_APPLEJPEGXL = $(WK_USE_APPLEJPEGXL_$(WK_PLATFORM_NAME));
WK_USE_APPLEJPEGXL_macosx = $(WK_USE_APPLEJPEGXL$(WK_MACOS_1400));
WK_USE_APPLEJPEGXL_MACOS_SINCE_1400 = YES;
WK_USE_APPLEJPEGXL_MACOS_BEFORE_1400 = NO;
```

hortont424 replied:
Pretty sure the current patch is correct (we don't want to use AppleJPEGXL on Sonoma, see https://bugs.webkit.org/show_bug.cgi?id=257768)

brentfulgham replied:
If Tim's satisfied, so am I. Adding merge-queue so we can move on!

litherum committed:
[macOS] Temporarily disable JPEG XL on older OSes until we can get https://bugs.webkit.org/show_bug.cgi?id=259587 ironed out
https://bugs.webkit.org/show_bug.cgi?id=259603
rdar://113041105

Reviewed by Tim Horton.

This path is producing images that are garbled. Temporarily disable the feature until we can fix that.

* Source/WebCore/Configurations/WebCore.xcconfig:

Canonical link: https://commits.webkit.org/266408@main
